Test build fails with kernel < 3.4?

On CentOS 6 (kernel-headers-2.6.32):

```
$ make test
cc -Wall -Werror -D_GNU_SOURCE -g   -c -o test/victim.o test/victim.c
test/victim.c: In function ‘main’:
test/victim.c:8: error: ‘PR_SET_PTRACER’ undeclared (first use in this function)
test/victim.c:8: error: (Each undeclared identifier is reported only once
test/victim.c:8: error: for each function it appears in.)
test/victim.c:8: error: ‘PR_SET_PTRACER_ANY’ undeclared (first use in this function)
make: *** [test/victim.o] Error 1
```

I suppose PR_SET_PTRACER was added in kernel 3.4. Not sure what can be done about this, but if nothing else, document it?

Yes, that fixes the compilation. The actual tests still end up failing:

```
$ make test
python test/basic.py
python test/tty-steal.py
Traceback (most recent call last):
  File "test/tty-steal.py", line 12, in <module>
    prctl.set_ptracer(PR_SET_PTRACER_ANY)
AttributeError: 'module' object has no attribute 'set_ptracer'
make: *** [test] Error 1
```
But I suppose that prctl stuff isn't needed either? If so, would be nice if the prctl python module would be required only wherever its set_ptracer would do any good.

--- a/test/tty-steal.py
+++ b/test/tty-steal.py
@@ -6,11 +6,13 @@ if os.getenv("NO_TEST_STEAL") is not None:
     print("Skipping tty-stealing tests because $NO_TEST_STEAL is set.")
     sys.exit(0)
 
-import prctl
-
-PR_SET_PTRACER_ANY = 0xffffffff
-if hasattr(prctl, 'set_ptracer'):
-    prctl.set_ptracer(PR_SET_PTRACER_ANY)
+try:
+    import prctl
+    PR_SET_PTRACER_ANY = 0xffffffff
+    if hasattr(prctl, 'set_ptracer'):
+        prctl.set_ptracer(PR_SET_PTRACER_ANY)
+except ImportError:
+    print("Unable to import `prctl`, skipping `PR_SET_PTRACER`.")
 
 child = pexpect.spawn("test/victim")
 child.setecho(False)

--- a/test/tty-steal.py
+++ b/test/tty-steal.py
@@ -9,7 +9,8 @@ if os.getenv("NO_TEST_STEAL") is not None:
 import prctl
 
 PR_SET_PTRACER_ANY = 0xffffffff
-prctl.set_ptracer(PR_SET_PTRACER_ANY)
+if hasattr(prctl, 'set_ptracer'):
+    prctl.set_ptracer(PR_SET_PTRACER_ANY)
 
 child = pexpect.spawn("test/victim")
 child.setecho(False)

--- a/test/victim.c
+++ b/test/victim.c
@@ -5,7 +5,9 @@ int main(int argc, char **argv) {
     char *line = NULL;
     size_t cap = 0;
 
+#ifdef PR_SET_PTRACER
     prctl(PR_SET_PTRACER, PR_SET_PTRACER_ANY);
+#endif
 
     while(getline(&line, &cap, stdin) != -1) {
         printf("ECHO: %s", line);

